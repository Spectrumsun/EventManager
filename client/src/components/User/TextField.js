import React from 'react';

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
      //required
    />
  </div>
);


export default TextField;
