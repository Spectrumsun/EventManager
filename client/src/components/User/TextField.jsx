import React from 'react';
import PropTypes from 'prop-types'; 

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
    
    />
  </div>
);


TextField.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default TextField;
