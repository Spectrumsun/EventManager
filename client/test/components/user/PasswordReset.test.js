import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import ConnectedPasswordReset,
{ PasswordReset } from '../../../src/components/User/PasswordReset';
import TextField from '../../../src/components/User/TextField'


const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  isAuthenticated: false,
  user: {},
};
const store = mockStore(initialState);

const props = {
  initpasswordreset: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory(),
  match: {
    params: {
      token: 'lkjhuiokj'
    }
  }
};

  const mountedWrapper = mount(
    <Provider store={store}>
    <BrowserRouter>
      <ConnectedPasswordReset {...props} />
    </BrowserRouter>
    </Provider>
  );

const shallowWrapper = shallow(<PasswordReset {...props} />);

const state = {
  password: '',
  confirmPassword: '',
};

const event = {
  preventDefault: jest.fn(),
  target: {
    password: '123456789',
    confirmPassword: '123456789',
  }
};


describe('<PasswordReset /> Component', () => {
  it('should render the <Login />', () => {
    shallow(<PasswordReset />);
  });

  it('should render the <PasswordReset /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('PasswordReset').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <PasswordReset {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });


  it('should render initial layout of PasswordReset', () => {
    const wrapper = shallow(<PasswordReset />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have three image on layout', () => {
    const wrapper = shallow(<PasswordReset />);
    expect(wrapper.find('div').length).toEqual(7);
  });

  it('should have three image on layout', () => {
    const wrapper = shallow(<PasswordReset />);
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should have <TextField /> when the page loads', () => {
    const wrapper = shallow(<PasswordReset />);
    expect(wrapper.find(TextField)).toHaveLength(2);
  });


  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });
});

