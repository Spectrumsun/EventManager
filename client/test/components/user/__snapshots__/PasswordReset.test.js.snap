// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PasswordReset /> Component should match component snapshot 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "paddingTop": "150px",
      }
    }
  >
    <h3
      className="center"
    >
      Event Manager
    </h3>
    <div
      className="card loginCard"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <div
        className="card-header"
      >
        <h3>
          Password Reset
        </h3>
      </div>
      <div
        className="card-body"
      >
        <div
          className="cont card-body"
        >
          <form
            className="centerform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                New Password
              </label>
              <input
                className="form-control form-control-lg"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Password
              </label>
              <input
                className="form-control form-control-lg"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirm Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="text-center"
            >
              <button
                className="btn btn-outline-dark"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PasswordReset /> Component should render initial layout of PasswordReset 1`] = `
Array [
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "150px",
        }
      }
    >
      <h3
        className="center"
      >
        Event Manager
      </h3>
      <div
        className="card loginCard"
        style={
          Object {
            "width": "30rem",
          }
        }
      >
        <div
          className="card-header"
        >
          <h3>
            Password Reset
          </h3>
        </div>
        <div
          className="card-body"
        >
          <div
            className="cont card-body"
          >
            <form
              className="centerform"
              onSubmit={[Function]}
            >
              <TextField
                label="New Password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <TextField
                label="Password"
                name="confirmPassword"
                onChange={[Function]}
                placeholder="confirm Password"
                type="password"
                value=""
              />
              <div
                className="text-center"
              >
                <button
                  className="btn btn-outline-dark"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
